src/boards/board_ttgo_t9_rgb_led_wm8978.h

```diff
@@ -30,41 +31,272 @@
 
 /**
  * @file board_ttgo_t9_rgb_led_wm8978.h
- * @author Marcel Licence
- * @date 20.12.2022
+ * @author Marcel Licence / David Agnér
+ * @date 02.05.2023
  *
- * @brief Board description for TTGO-T9-RGB_LED-WM8978
+ * @brief This file contains init stuff for LilyGO TTGO T9 (v1.5) module
  * @see https://github.com/LilyGO/TTGO-TAudio
+ * @see Schematic: https://github.com/LilyGO/TTGO-TAudio/blob/master/schematic.pdf
  */
+ 
+ 
+ /* ---------  Pin mapping -----------------
+ 
+ PIN	FUNCTION			Note
+  0		I2S_MCLK_PIN		DAC WM8978 /HEADER
+  1		TX UART0			CP210x USB 
+  2		SPI_MISO			SD-card /HEADER
+  3		RX UART0			CP210x USB
+  4 	BLINK_LED_PIN		HEADER
+  5		I2C_SCL_2			HEADER
+ 12		FREE				HEADER
+ 13		SPI_CS				SD-card /HEADER
+ 14		SPI_SCK				SD-card
+ 15		SPI_MOSI			SD-card /HEADER
+ 18		I2C_SCL				Control WM8978 
+ 19		I2C_SDA				Control WM8978
+ 22 	LED_STRIP_PIN		WS2812b LED strip
+ 23		I2C_SDA_2			HEADER
+ 25		I2S_WCLK_PIN		DAC WM8978
+ 26		I2S_DOUT_PIN		DAC WM8978
+ 27		I2S_DIN_PIN			ADC WM9878
+ 32		FREE				HEADER
+ 33		I2S_BCLK_PIN		DAC WM8978
+ 34		PIN_KEY_4			Button 4 /HEADER
+ 35		MIDI_RX2			MIDI in /HEADER
+ 36		PIN_KEY_3			Button 3
+ 39		PIN_KEY_2			Button 2
+ 
 
+ */
+#ifdef __CDT_PARSER__
+#include <cdt.h>
+#endif
 
 #ifndef BOARDS_BOARD_TTGO_T9_RGB_LED_WM8978_H_
 #define BOARDS_BOARD_TTGO_T9_RGB_LED_WM8978_H_
 
-
-/* on board led, green */
-#define BLINK_LED_PIN     22
+#include <Wire.h>
+#include <WM8978.h> // https://github.com/CelliesProjects/wm8978-esp32
+#include <FastLED.h> // https://github.com/FastLED/FastLED
 
 #define MIDI_PORT2_ACTIVE
-#define MIDI_RX2_PIN 34 /* use a free pin! */
+#define MIDI_RX2_PIN 35 /* Free on HEADER */
 
-#define LED_STRIP_PIN 22 /* onboard LEDs */
 
+/* on board led, green */
+//#define BLINK_LED_PIN     22
+#define BLINK_LED_PIN     4 /* use external free pin on header */
 
-#define I2C_SDA 19
-#define I2C_SCL 18
 
+/* For WS2812B leds */
+#define LED_STRIP_PIN 22 /* onboard LEDs */
+#define NUM_LEDS      19
+#define BRIGHTNESS    64
+#define LED_TYPE      WS2812B
+#define COLOR_ORDER   RGB
 
-//#define I2C_SDA 23
-//#define I2C_SCL 5
+/* The WM9878 DAC needs I2C for setup but it is not on the HEADER pins, use second I2C bus for (wire2) for OLED and ADS1115 for pots ? */
+#define I2C_SDA_2 23		// For OLED later on ?
+#define I2C_SCL_2 5		  // For OLED later on ?
 
 
 /* WM8978 */
+#define I2C_SDA 19		/* To control WM8978 */
+#define I2C_SCL 18		/* To control WM8978 */
 #define I2S_MCLK_PIN 0 /* 24 MHz */
```

Comment:
24 MHz fixed or depending on bitrate?

Reply:
Haven't got a clue. I used the default from the arduino lib